jupyter_client/session.py

```diff
@@ -92,12 +93,27 @@ def squash_unicode(obj):
 
 
 def json_packer(obj):
-    return json.dumps(
-        obj,
-        default=json_default,
-        ensure_ascii=False,
-        allow_nan=False,
-    ).encode("utf8")
+    try:
+        return json.dumps(
+            obj,
+            default=json_default,
+            ensure_ascii=False,
+            allow_nan=False,
+        ).encode("utf8")
+    except ValueError as e:
+        # Fallback to trying to clean the json before serializing
+        warnings.warn(
```

Comment:
Maybe show this warning only after the second try at serialization post json_clean succeeds? It might get confusing to get a deprecation warning if it really should fail.